src/main/java/uk/gov/hmcts/reform/blobrouter/reconciliation/service/SummaryReportService.java

```diff
@@ -0,0 +1,146 @@
+import uk.gov.hmcts.reform.blobrouter.config.ServiceConfiguration;
+import uk.gov.hmcts.reform.blobrouter.config.StorageConfigItem;
+import uk.gov.hmcts.reform.blobrouter.config.TargetStorageAccount;
+import uk.gov.hmcts.reform.blobrouter.data.envelopes.Envelope;
+import uk.gov.hmcts.reform.blobrouter.data.reconciliation.reports.ReconciliationReportRepository;
+import uk.gov.hmcts.reform.blobrouter.data.reconciliation.reports.model.NewReconciliationReport;
+import uk.gov.hmcts.reform.blobrouter.data.reconciliation.statements.SupplierStatementRepository;
+import uk.gov.hmcts.reform.blobrouter.data.reconciliation.statements.model.EnvelopeSupplierStatement;
+import uk.gov.hmcts.reform.blobrouter.reconciliation.model.in.SupplierStatement;
+import uk.gov.hmcts.reform.blobrouter.reconciliation.report.SummaryReport;
+import uk.gov.hmcts.reform.blobrouter.reconciliation.report.SummaryReportItem;
+import uk.gov.hmcts.reform.blobrouter.services.EnvelopeService;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static java.util.Collections.emptyList;
+import static java.util.stream.Collectors.groupingBy;
+import static org.slf4j.LoggerFactory.getLogger;
+
+@Service
+public class SummaryReportService {
+
+    private static final Logger logger = getLogger(SummaryReportService.class);
+
+    private final SupplierStatementRepository repository;
+    private final ReconciliationReportRepository reconciliationReportRepository;
+    private final ObjectMapper objectMapper;
+    private final EnvelopeService envelopeService;
+    private final Map<String, StorageConfigItem> storageConfig; // container-specific configuration, by container name
+
+    public SummaryReportService(
+        SupplierStatementRepository repository,
+        ReconciliationReportRepository reconciliationReportRepository,
+        ObjectMapper objectMapper,
+        EnvelopeService envelopeService,
+        ServiceConfiguration serviceConfiguration
+    ) {
+        this.repository = repository;
+        this.reconciliationReportRepository = reconciliationReportRepository;
+        this.objectMapper = objectMapper;
+        this.envelopeService = envelopeService;
+        this.storageConfig =  serviceConfiguration.getStorageConfig();
+    }
+
+    public void process(LocalDate date) throws JsonProcessingException {
+        Optional<EnvelopeSupplierStatement> optSupplierStatement = repository.findLatest(date);
+
+        if (!optSupplierStatement.isPresent()) {
+            logger.error("No supplier statement found for {}", date);
+            return;
+        }
+
+        EnvelopeSupplierStatement envelopeSupplierStatement = optSupplierStatement.get();
+        SupplierStatement supplierStatement = objectMapper
+            .readValue(envelopeSupplierStatement.content, SupplierStatement.class);
+
+        var envelopeList = envelopeService.getEnvelopes(date);
```

Comment:
Unfortunately when envelope gets rejected, no envelope is created in DB. We can't use this.

Reply:
I see rejected enveloped in the table.